src/Workspaces/Core/Portable/Workspace/Host/PersistentStorage/IPersistentStorageConfiguration.cs

```diff
@@ -39,20 +39,25 @@ internal sealed class DefaultPersistentStorageConfiguration : IPersistentStorage
         /// </summary>
         private static readonly ImmutableArray<char> s_invalidPathChars = Path.GetInvalidPathChars().Concat('/').ToImmutableArray();
 
-        [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public DefaultPersistentStorageConfiguration()
-        {
-        }
+        private static readonly string s_cacheDirectory;
+        private static readonly string s_moduleFileName;
 
-        private static string GetCacheDirectory()
+        static DefaultPersistentStorageConfiguration()
         {
             // Store in the LocalApplicationData/Roslyn/hash folder (%appdatalocal%/... on Windows,
             // ~/.local/share/... on unix).  This will place the folder in a location we can trust
             // to be able to get back to consistently as long as we're working with the same
             // solution and the same workspace kind.
             var appDataFolder = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData, Environment.SpecialFolderOption.Create);
-            return Path.Combine(appDataFolder, "Microsoft", "VisualStudio", "Roslyn", "Cache");
+            s_cacheDirectory = Path.Combine(appDataFolder, "Microsoft", "VisualStudio", "Roslyn", "Cache");
+
+            s_moduleFileName = SafeName(Process.GetCurrentProcess().MainModule.FileName);
```

Comment:
in particular, .MainModule turns out to be extremely expensive, and it looks like it is expensive every time it is run.  Given that we keep creating fresh storage instances (also addressed here https://github.com/dotnet/roslyn/pull/57513) this turned out to be a huge hotspot.